commands/curl.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package commands
+
+import (
+	"fmt"
+	"odfe-cli/client"
+	ctrl "odfe-cli/controller/es"
+	entity "odfe-cli/entity/es"
+	gateway "odfe-cli/gateway/es"
+	handler "odfe-cli/handler/es"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	curlCommandName              = "curl"
+	curlPrettyFlagName           = "pretty"
+	curlPathFlagName             = "path"
+	curlQueryParamsFlagName      = "query-params"
+	curlDataFlagName             = "data"
+	curlHeadersFlagName          = "headers"
+	curlOutputFormatFlagName     = "output-format"
+	curlOutputFilterPathFlagName = "filter-path"
+)
+
+//curlCommand is base command for Elasticsearch REST APIs.
+var curlCommand = &cobra.Command{
+	Use:   curlCommandName,
+	Short: "Manage elasticsearch core features",
+	Long:  "Use curl command to configure and access features directly",
+}
+
+func init() {
+	curlCommand.Flags().BoolP("help", "h", false, "Help for curl command")
+	curlCommand.PersistentFlags().Bool(curlPrettyFlagName, false, "Response will be formatted")
+	curlCommand.PersistentFlags().StringP(curlOutputFormatFlagName, "o", "",
+		"Output format if supported by cluster, else, default format by Elasticsearch. Example json, yaml")
```

Comment:
Im not really sure what this means. Could you elaborate?

Reply:
Elasticsearch support outputs in json/yaml/table format. Similar to how "Accept-header" works. Unfortunately, not all api support all formats. For ex: _cluster/health gives only json/yaml, while _cat/indices support all 3. Hence, if user wants some specific format, they can provide this option but it will be only honored by server if it can support.